Source/Tools/FEXLoader/LinuxSyscalls/GdbServer.cpp

```diff
@@ -127,8 +127,12 @@ void GdbServer::WaitForThreadWakeup() {
 }
 
 GdbServer::~GdbServer() {
+  CloseListenSocket();
```

Comment:
Since OpenListenSocket is running at the start of the GDB server thread, is there potential for a race condition for short-lived FEX processes here?

Reply:
Not currently since gdbserver always forces processes to wait for gdb attach currently.

Reply:
Does it? I see `GDBServer::GDBServer()` ending in a call to StartThread, so the caller can immediately close it after. Is there external logic that does this waiting?

Reply:
> Is there external logic that does this waiting?
Yes.
https://github.com/FEX-Emu/FEX/blob/main/FEXCore/Source/Interface/Core/Core.cpp#L331

Reply:
Fair enough, thanks for clarifying!